src/EnergyPlus/DataPlant.hh

```diff
@@ -597,6 +601,7 @@ namespace DataPlant {
 		int IndexInLoopSidePumps; // If I'm a pump, this tells my index in PL(:)%LS(:)%Pumps
 		Real64 TempDesCondIn;
 		Real64 TempDesEvapOut;
+		std::shared_ptr< PlantComponent > compPtr;
```

Comment:
I'm going to take an unfiltered pass through this PR. Most of what I say will not be well thought out. I'm just blurting out my first impressions. 

shared_ptr ??? unique_ptr or something might be a better choice. Somebody should probably take responsibility for these objects. 

This also feels like a problem that is best solved up a level above Plant, but doing that might blow up your problem even more and in a way you are not ready to deal with.